### PR TITLE
Add account CTA

### PR DESCRIPTION
**Pull request by OKendigelyan**
## Proposed changes

[Task link](https://app.asana.com/0/1207041888417083/1207811126378442/f)

## Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor
- [ ] Breaking change
- [ ] UI fix

## Screenshots

Add the screenshots of how the app used to look like and how it looks now

image
image
image
image

## Checklist

- [ ] Tests that prove my fix is effective or that my feature works have been added
- [ ] Documentation has been added (if appropriate)
- [x] Screenshots are added (if any UI changes have been made)
- [ ] All TODOs have a corresponding task created (and the link is attached to it)

### CODE REVIEW

File: apps/web/src/components/AccountSelectorModal/RemoveAccountModal.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (isLastImplicitAccount) {
+      onClose();
+      navigate("/");
+    } else {
+      goBack();
+    }
+  };
+
+  let description =
```
Comment (asiia-trilitech) [inline apps/web/src/components/AccountSelectorModal/RemoveAccountModal.tsx:40]: I think Figma has slightly different description & button text: https://www.figma.com/design/kUJJ7T75lP4GPNMg4GMVt1/Embed-%26-Web?node-id=595-22299&t=tk0S3nUQLcCw7EUN-0

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+type AccountInfoModalProps = {
+  account: Account;
+};
+
+export const AccountInfoModal = ({ account }: AccountInfoModalProps) => {
+  const { goBack } = useDynamicDisclosureContext();
+
+  const handleCopyAddress = async (event: MouseEvent<HTMLButtonElement>) => {
+    await navigator.clipboard.writeText(account.address.pkh);
+    goBack();
```
Comment (asiia-trilitech) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx:28]: Should the modal get closed on copying the address? 
On desktop in similar cases we show toast and keep the modal open

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        <RemoveAccountModal account={account as SocialAccount | LedgerAccount | SecretKeyAccount} />
+      );
+    },
+    isEnabled: (account: Account) => account.type !== "mnemonic" && account.type !== "multisig",
+  },
+  {
+    icon: <ExternalLinkIcon />,
+    text: "View in TzKT",
+    action: (account: Account) => {
+      window.open(`https://tzkt.io/${account.address.pkh}`, "_blank");
```
Comment (asiia-trilitech) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx:62]: It's opening mainnet atm - can we open tzkt with the selected network?

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  };
+
+  return (
+    <ModalContent>
+      <ModalHeader>
+        <VStack gap="12px">
+          <Heading size="md">Account Info</Heading>
+          <Text maxWidth="340px" color="gray.700" fontWeight="400" size="md">
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx:36]: please use the useColor where possible

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          </Text>
+        </VStack>
+        <ModalCloseButton />
+      </ModalHeader>
+      <ModalBody>
+        <VStack gap="30px">
+          <Box
+            padding="30px"
+            background="white"
+            borderWidth={1.5}
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx:47]: please use the actual units

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          size="xs"
+        />
+      </PopoverTrigger>
+      <Portal>
+        <PopoverContent maxWidth="204px">
+          <PopoverBody color="gray.400">
+            {accountSelectorPopoverOptions.map(
+              ({ icon, text, action, isEnabled }, index) =>
+                isEnabled(account) && (
+                  <Flex
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx:102]: this should be a button. maybe with a separate variant

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+import { RenameAccountPage } from "./RenameAccountModal";
+import {
+  EditIcon,
+  ExternalLinkIcon,
+  SearchIcon,
+  ThreeDotsIcon,
+  TrashIcon,
+} from "../../assets/icons";
+
+const accountSelectorPopoverOptions = [
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx:31]: I think it'd be much cleaner if you just inlined it

File: apps/web/src/components/Menu/Menu.tsx [hunk 8]
```diff
@@ -21,21 +21,23 @@ import {
...
+  const navigate = useNavigate();
 
   const { isOpen, onOpen, onClose } = useDisclosure();
   const cancelRef = useRef(null);
 
   const handleLogout = () => {
     persistor.pause();
     localStorage.removeItem("persist:accounts");
-    window.location.replace("/");
+    navigate("/");
```
Comment (serjonya-trili) [inline apps/web/src/components/Menu/Menu.tsx:40]: with this, it's likely that the persistor will still be paused which is not exactly what we expect

**Review by serjonya-trili — CHANGES_REQUESTED** [review]
1. Copy address button should show a "Copied" popover instead of coming back
2. When a sub-modal is closed, it should return user back to the previous modal (e.g. open account selector)
3. Open on Tzkt should work for ghostnet too. right now it always opens the mainnet

**Comment by OKendigelyan** [comment]
> 2. When a sub-modal is closed, it should return user back to the previous modal (e.g. open account selector)
will be implemented in this ticket https://app.asana.com/0/1207041888417083/1207893626622358/f

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import { ModalCloseButton } from "../ModalCloseButton";
+
+type AccountInfoModalProps = {
+  account: Account;
+};
+
+export const AccountInfoModal = ({ account }: AccountInfoModalProps) => {
+  const color = useColor();
+  const { onOpen, isOpen } = useDisclosure();
+  const handleCopyAddress = async () => {
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.tsx:31]: maybe shorten to:

```typescript
const handleCopyAddress = () => navigator.clipboard.writeText(account.address.pkh);
```
?

File: apps/web/src/styles/theme.ts [hunk 10]
```diff
@@ -159,6 +159,18 @@ const theme = extendTheme({
         },
       },
       variants: {
+        dropdownOption: {
+          display: "flex",
+          justifyContent: "flex-start",
+          w: "full",
```
Comment (serjonya-trili) [inline apps/web/src/styles/theme.ts:165]: please avoid using these shorthands. it's too easy to make a mistake with those.

File: apps/web/src/assets/icons/index.ts [hunk 11]
```diff
@@ -29,9 +29,15 @@ export { default as TwitterIcon } from "./twitter.svg";
...
+export { default as EditIcon } from "./edit.svg";
+export { default as TrashIcon } from "./trash.svg";
+export { default as ExternalLinkIcon } from "./external-link.svg";
+export { default as SearchIcon } from "./search.svg";
+export { default as CloseIcon } from "./close.svg";
+export { default as AlertIcon } from "./alert.svg";
 export { TokenIcon } from "./TokenIcon";
 
-import CrossIcon from "./cross.svg";
+import CloseIcon from "./close.svg";
```
[the remark this answers is not in the document]
Comment (OKendigelyan) [reply apps/web/src/assets/icons/index.ts:40]: `close.svg` is relevant icon from the figma

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    expect(screen.getByText("Test Account")).toBeInTheDocument();
+    expect(screen.getByText("tz1TestAddress")).toBeInTheDocument();
+  });
+
+  it("displays 'Copied!' message in popover when button is clicked", async () => {
+    const user = userEvent.setup();
+
+    render(<AccountInfoModal account={mockAccount} />);
+
+    await act(async () => {
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx:30]:
```typescript
await act(() => user.click(screen.getByRole("button", { name: /copy wallet address/i })));
```
is shorter

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx [hunk 13]
```diff
@@ -0,0 +1,36 @@
+import { type ImplicitAccount } from "@umami/core";
+
+import { AccountInfoModal } from "./AccountInfoModal";
+import { act, render, screen, userEvent } from "../../testUtils";
+
+const mockAccount = {
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx:6]: we have helpers for this. like mockImplicitAccount

File: apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+  label: "Test Account",
+  address: { pkh: "tz1TestAddress" },
+} as ImplicitAccount;
+
+describe("<AccountInfoModal />", () => {
+  it("renders correctly with initial values", () => {
+    render(<AccountInfoModal account={mockAccount} />);
+
+    expect(screen.getByText("Account Info")).toBeInTheDocument();
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountInfoModal.test.tsx:15]: it's better to use `toBeVisible()`, and in other places too

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+  type: "social",
+} as ImplicitAccount;
+
+const mockNetwork = {
+  tzktExplorerUrl: "https://tzkt.io",
+};
+
+describe("AccountSelectorPopover", () => {
+  beforeEach(() => {
+    (useDynamicDisclosureContext as jest.Mock).mockReturnValue({
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx:39]: it's already available to you with the `dynamicDisclosureContextMock`. please check testUtils.tsx

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+      openWith: jest.fn(),
+    });
+    (useSelectedNetwork as jest.Mock).mockReturnValue(mockNetwork);
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx:42]: it's better to use mainnet VS ghostnet instead

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+    });
+
+    expect(screen.getByText("Account Info")).toBeInTheDocument();
+    expect(screen.getByText("Rename")).toBeInTheDocument();
+    expect(screen.getByText("Remove")).toBeInTheDocument();
+    expect(screen.getByText("View in TzKT")).toBeInTheDocument();
+  });
+
+  it("opens AccountInfoModal when 'Account Info' button is clicked", async () => {
+    const { openWith } = useDynamicDisclosureContext();
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.test.tsx:66]: please use `dynamicDisclosureContextMock` instead

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+                }}
+                variant="dropdownOption"
+              >
+                <SearchIcon />
+                <Text color={color("900")} fontWeight="600">
+                  Account Info
+                </Text>
+              </Button>
+              <Button
+                onClick={async e => {
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx:72]: nit: there is no need for async await.

you can just `return openWith(...)`
(and in other places too)

File: apps/web/src/components/AccountSelectorModal/RenameAccountModal.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+  };
+
+  const validateName = useValidateName(account.label);
+
+  return (
+    <ModalContent>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <ModalHeader>
+          <VStack gap="18px">
+            <EditIcon width="22px" color={light.blue} />
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/RenameAccountModal.tsx:57]: why not `color={color("blue")}` ?

**Review by serjonya-trili — CHANGES_REQUESTED** [review]

File: apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+                    Remove
+                  </Text>
+                </Button>
+              )}
+              <Button
+                onClick={e => {
+                  e.stopPropagation();
+
+                  const url = `${network.tzktExplorerUrl}/${account.address.pkh}`;
+                  window.open(url, "_blank");
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx:106]: this is a security issue https://docs.datadoghq.com/code_analysis/static_analysis_rules/jsx-react/jsx-no-target-blank/

please use a `<Link isExternal` instead. [Example](https://github.com/trilitech/umami-v2/pull/1650/files#diff-ada356f9b19ff984f99256292da9132e180abdb48ab4db214f6b084d9429b155R49)

File: apps/web/src/components/AccountSelectorModal/RemoveAccountModal.test.tsx [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders with default description and button label", async () => {
+    (useImplicitAccounts as jest.Mock).mockReturnValue([{}, {}]);
+
+    render(<RemoveAccountModal account={{} as any} />);
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/RemoveAccountModal.test.tsx:33]: and if you do prepare the state, you wouldn't need to pass this meaningless value here

**Review by serjonya-trili — CHANGES_REQUESTED** [review]

File: apps/web/src/components/AccountSelectorModal/RenameAccountModal.test.tsx [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+  it("dispatches renameAccount and navigates back on valid form submission", async () => {
+    const user = userEvent.setup();
+    const { goBack } = dynamicDisclosureContextMock;
+
+    render(<RenameAccountPage account={mockAccount} />);
+
+    await act(() => user.clear(screen.getByPlaceholderText("Enter contact's name")));
+    await act(() => user.type(screen.getByPlaceholderText("Enter contact's name"), "New Name"));
+    await act(() => user.click(screen.getByRole("button", { name: /update/i })));
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/RenameAccountModal.test.tsx:48]: please use exact strings where possible

File: apps/web/src/components/AccountSelectorModal/RemoveAccountModal.test.tsx [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+jest.mock("@umami/state", () => ({
+  ...jest.requireActual("@umami/state"),
+  useRemoveAccount: jest.fn(),
+}));
+
+describe("<RemoveAccountModal />", () => {
+  const mockRemoveAccount = jest.fn();
+
+  beforeEach(() => {
+    (useRemoveAccount as jest.Mock).mockReturnValue(mockRemoveAccount);
```
Comment (serjonya-trili) [inline apps/web/src/components/AccountSelectorModal/RemoveAccountModal.test.tsx:37]: please use `jest.mocked(useRemoveAccount)` instead

**Review by serjonya-trili — APPROVED** [review]

**Pull request merged**